.github/workflows/windows1.yml

```diff
@@ -0,0 +1,115 @@
+name: Windows tests (MSVC + ifx + OpenBLAS)
+
+on:
+  push:
+    branches:
+      - maintenance/**
+  pull_request:
+    branches:
+      - main
+      - maintenance/**
+
+permissions:
+   contents: read  # to fetch code (actions/checkout)
+
+env:
+  WINDOWS_BASEKIT_URL: https://registrationcenter-download.intel.com/akdlm/IRC_NAS/7dff44ba-e3af-4448-841c-0d616c8da6e7/w_BaseKit_p_2024.1.0.595_offline.exe
+  WINDOWS_HPC_URL: https://registrationcenter-download.intel.com/akdlm/IRC_NAS/c95a3b26-fc45-496c-833b-df08b10297b9/w_HPCKit_p_2024.1.0.561_offline.exe
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  get_commit_message:
+    name: Get commit message
+    uses: ./.github/workflows/commit_message.yml
+
+  msvc_ifx_blas:
+    name: py3.11, dev.py
+    needs: get_commit_message
+    # Ensure (a) this doesn't run on forks by default, and
+    #        (b) it does run with Act locally (`github` doesn't exist there)
+    if: >
+      needs.get_commit_message.outputs.message == 1
+      && (github.repository == 'scipy/scipy' || github.repository == '')
+    runs-on: windows-2022
+    defaults:
+      run:
+        shell: powershell
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          submodules: recursive
+
+      - name: Setup conda
+        uses: s-weigand/setup-conda@v1
+        with:
+          update-conda: true
+          python-version: 3.11
+      - run: conda --version
+      - run: which python
+
+      - name: Install packages from conda
+        run: |
+          conda install -c conda-forge pkg-config meson=1.4.0 meson-python=0.16.0 ninja openblas libblas=*=*openblas
+
+      - name: pip-packages
+        run: |
+          pip install numpy==1.23.5 cython pybind11 pytest pytest-xdist pytest-timeout pytest-fail-slow pooch rich_click click doit pydevtool hypothesis
+
+      - name: cache install
+        id: cache-install
+        uses: actions/cache@v3
+        with:
+          path: |
+              C:\Program Files (x86)\Intel\oneAPI\compiler
+          key: install-${{ env.WINDOWS_HPC_URL }}-${{ env.WINDOWS_BASEKIT_URL }}-compiler
+      - name: Install oneAPI Base kit
+        if: steps.cache-install.outputs.cache-hit != 'true'
+        run: |
+          echo %WINDOWS_BASEKIT_URL%
+          tools/install_intel_oneAPI_windows.bat %WINDOWS_BASEKIT_URL%
+      - name: Install oneAPI HPC kit
+        if: steps.cache-install.outputs.cache-hit != 'true'
+        run: |
+          echo %WINDOWS_HPC_URL%
+          tools/install_intel_oneAPI_windows.bat %WINDOWS_HPC_URL%
+
+      # The following steps are necessary because calling `.bat` files for Intel oneAPI and
+      # MSVC doesn't update the environment on CI. Both `ifx` and `cl` aren't detected resulting
```

Comment:
What updates to the environment do you expect? I would like to explore some more if we can do this programmatically somehow, because the below is very fragile.

Reply:
> because the below is very fragile.

I agree 100% with you. For activating `ifx`, ideal way would be too call, `C:\"Program Files (x86)"\Intel\oneAPI\setvars.bat`. On my Azure machine it works. However, this doesn't work on CI. `ifx` command doesn't work. So I had to manually update `PATH` and `LIB` for `ifx`. This is the only way to make `ifx` detected. This further causes `cl` to stop working. So I had to update, `PATH`, `LIB` and `INCLUDE`. This made `cl` work. You can see all the attempts in https://github.com/czgdp1807/scipy/pull/1.

Reply:
I have thought of some pragmatic ways to do this. Let me try those in my own fork and then push the final changes here.